var/spack/repos/builtin/packages/py-h5py/package.py

```diff
@@ -42,12 +43,13 @@ class PyH5py(PythonPackage):
     depends_on("python@3.7:", type=("build", "run"), when="@3.2:")
 
     # Build dependencies
-    depends_on("py-cython@0.23:", type="build", when="@:2")
-    depends_on("py-cython@0.29:", type=("build"), when="@3: ^python@:3.7")
-    depends_on("py-cython@0.29.14:", type=("build"), when="@3: ^python@3.8.0:3.8")
-    depends_on("py-cython@0.29.15:", type=("build"), when="@3: ^python@3.9.0:")
+    depends_on("py-cython@0.23:0", type="build", when="@:2")
+    depends_on("py-cython@0.29:0", type=("build"), when="@3: ^python@:3.7")
+    depends_on("py-cython@0.29.14:0", type=("build"), when="@3: ^python@3.8.0:3.8")
+    depends_on("py-cython@0.29.15:0", type=("build"), when="@3: ^python@3.9.0:")
     depends_on("py-pkgconfig", type="build")
     depends_on("py-setuptools", type="build")
+    # depends_on("py-setuptools@61:", type="build", when="@3.8.0:")
```

Comment:
I would uncomment this since it's an actual requirement. If it causes conflicts with other packages, then those packages will simply use an older versions of h5py. @alalazo and others are working on allowing the same package to appear in the DAG multiple times.

Reply:
Yup, building with an older setuptools can certainly cause issues - we discovered this recently when a bug in pip meant it used incorrect build dependencies in some situations: https://github.com/h5py/h5py/issues/2240

Reply:
well the reason was to placate me so the data-vis-sdk and e4s stacks can build with vol adaptors...but as of right now that is lower priority until May, do you have any ideas when duplicate build deps will be allowed?

@alalazo

Reply:
But data-vis-sdk and e4s stacks can build with vol adaptors... using h5py 3.7.0. Or is that a feature only present in newer hdf5 supported in h5py 3.8.0? 

Either way, I don't think it's possible to support what you want until Spack can resolve multiple build deps in the DAG at the same time. As @takluyver mentioned, the 61+ requirement isn't just cosmetic, the package does not reliably build with old setuptools.

Reply:
h5py 3.7.0 does not support HDF5@1.13.2: which is a hard requirement for the VOLS, they don't exist before that. And the SDK heavily prefers the HDF5 1.14.0 since that is the actual release version, the odd versions are experimental.

Reply:
Is there a reason for packages to _not_ use a newer setuptools? I am not a heavy python user, I usually just use the latest pip gives me.

Reply:
I've pushed a new version that now also includes the setuptools@61: dependency.

Reply:
> Is there a reason for packages to not use a newer setuptools?

Some packages use features added in newer versions of setuptools, while other packages use features that were removed in newer versions of setuptools (e.g., 2to3). You may be able to reach out to the developers of the latter packages and ask for support for newer setuptools. Sometimes the documented dependency is incorrect and newer versions are actually supported.